client/app/scripts/superdesk-templates/templates.js

```diff
@@ -116,37 +126,114 @@
         };
     }
 
-    var app = angular.module('superdesk.templates', ['superdesk.authoring']);
-    app.service('templatesSettingsService', TemplatesSettingsService)
-        .directive('sdTemplates', TemplatesDirective);
-
-    app
-        .config(['superdeskProvider', function(superdesk) {
-            superdesk
-                .activity('/settings/templates', {
-                    label: gettext('Templates'),
-                    templateUrl: 'scripts/superdesk-templates/views/settings.html',
-                    controller: TemplatesSettingsController,
-                    category: superdesk.MENU_SETTINGS,
-                    privileges: {content_templates: 1},
-                    priority: 2000,
-                    beta: true
-                });
-        }])
-        .config(['apiProvider', function(apiProvider) {
-            apiProvider.api('templates', {
-                type: 'http',
-                backend: {
-                    rel: 'templates'
-                }
+    CreateTemplateController.$inject = ['item', 'templates', 'api', 'desks', '$q'];
+    function CreateTemplateController(item, templates, api, desks, $q) {
+        var vm = this,
+            metadata = [
+                'headline',
+                'slugline',
+                'abstract',
+                'dateline',
+                'byline',
+                'usage_terms',
+                'subject',
+                'genre',
+                'type',
+                'language',
+                'anpa-category',
+                'anpa_take_key',
+                'keywords',
+                'priority',
+                'urgency',
+                'pubstatus',
+                'description',
+                'body_html',
+                'body_text',
+                'place',
+                'located',
+                'creditline'
+            ];
+
+        this.type = 'create';
+        this.name = item.slugline || null;
+        this.desk = desks.active.desk || null;
+
+        this.types = templates.types;
+        this.save = save;
+
+        activate();
+
+        function activate() {
```

Comment:
Since this `activate()` function is invoked only once at controller instantiation and is not visible to the outside world, it might make sense to get rid of it and simply only retain its body?